extensions/walletProviders/utxo/src/extension.spec.ts

```diff
@@ -0,0 +1,126 @@
+
+  const path = "m/44'/0'/0'/0"
+  const mnemonic =
+    'audit hamster slot gesture brave fork ability agree over tiger bread soap bullet squeeze slide case roof drip seven plunge yard cat ozone custom'
+  const xpub =
+    'tpubDE23Kfj1twxmcCYAzuhAtw7se4xVdu9z6R1Psykd5HLyLtNbx4MFSsqi47FRg6PsQQJUrFUNUUViZiPUnmTthvd27idX2rorToRvGar5e9A'
+  const privateKey = 'cVeQk2DzVp8q5AVz4VS1vTqAGn7BmdB7SjrvtB3HPMFP8rpQeeZn'
+  const address = 'muNjVhqHUfNzHBrwawaGmjBt9suoMr1YxV'
+
+  describe('generateMnemonic', () => {
+    it('should generate 24 word mnemonic', () => {
+      const result = tatumSdk.walletProvider.use(UtxoWalletProvider).generateMnemonic()
+      expect(result.split(' ')).toHaveLength(24)
+    })
+  })
+  describe('generateXpub', () => {
+    it('should generate xpub', async () => {
+      const result = await tatumSdk.walletProvider.use(UtxoWalletProvider).generateXpub()
+      expect(result.xpub).toBeTruthy()
+      expect(result.mnemonic).toBeTruthy()
+      expect(result.derivationPath).toBe(path)
+    })
+  })
+
+  describe('generateXpub', () => {
+    it('should generate xpub from mnemonic', async () => {
+      const result = await tatumSdk.walletProvider.use(UtxoWalletProvider).generateXpub(mnemonic)
+      expect(result.xpub).toBe(xpub)
+      expect(result.mnemonic).toBe(mnemonic)
+      expect(result.derivationPath).toBe(path)
+    })
+  })
+
+  describe('generatePrivateKeyFromMnemonic', () => {
+    it('should generate private key', async () => {
+      const result = await tatumSdk.walletProvider
+        .use(UtxoWalletProvider)
+        .generatePrivateKeyFromMnemonic(mnemonic, 0)
+      expect(result).toBe(privateKey)
+    })
+  })
+
+  describe('generateAddressFromMnemonic', () => {
+    it('should generate address', async () => {
+      const result = await tatumSdk.walletProvider
+        .use(UtxoWalletProvider)
+        .generateAddressFromMnemonic(mnemonic, 0)
+      expect(result).toBe(address)
+    })
+  })
+
+  describe('generateAddressFromXpub', () => {
+    it('should generate address from xpub', () => {
+      const result = tatumSdk.walletProvider.use(UtxoWalletProvider).generateAddressFromXpub(xpub, 0)
+      expect(result).toBe(address)
+    })
+  })
+
+  describe('generateAddressFromPrivateKey', () => {
+    it('should generate address from private key', () => {
+      const result = tatumSdk.walletProvider.use(UtxoWalletProvider).generateAddressFromPrivateKey(privateKey)
+      expect(result).toBe(address)
+    })
+  })
+
+  describe('getWallet', () => {
+    it('should generate wallet', async () => {
+      const wallet = await tatumSdk.walletProvider.use(UtxoWalletProvider).getWallet()
+      expect(wallet.address).toBeTruthy()
+      expect(wallet.privateKey).toBeTruthy()
+      expect(wallet.mnemonic).toBeTruthy()
+      expect(wallet.mnemonic.split(' ')).toHaveLength(24)
+    })
+  })
+
+  describe('signAndBroadcast', () => {
+    it('should sign and broadcast transaction from address', async () => {
```

Comment:
tests with exposed pk fill be failing very soon :)

Reply:
Those are just dev tests, we don't have pipelines yet. Those addressed are empty